src/Workspaces/Remote/ServiceHub/Shared/Extensions.cs

```diff
@@ -19,6 +19,15 @@ namespace Microsoft.CodeAnalysis.Remote
 {
     internal static partial class Extensions
     {
+        private static readonly JsonRpcTargetOptions s_jsonRpcTargetOptions = new JsonRpcTargetOptions()
+        {
+            // Do not allow JSON-RPC to automatically subscribe to events and remote their calls.
+            NotifyClientOfEvents = false,
```

Comment:
can you explain what this option means?  i understand the one below, but not this one.

Reply:
Unless this is set, JSON RPC automatically (via reflection) subscribes to events defined on the target objects and remotes the call to the other side. We don't want that.